go/api/config.go

```diff
@@ -1,47 +1,71 @@
+/**
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
 package api
 
 import "github.com/aws/glide-for-redis/go/glide/protobuf"
 
 // NodeAddress represents the host address and port of a node in the cluster.
 type NodeAddress struct {
-	Host string
-	Port uint32
+	Host *string // Optional: if not supplied, "localhost" will be used.
+	Port *uint32 // Optional: if not supplied, 6379 will be used.
 }
 
 // RedisCredentials represents the credentials for connecting to a Redis server.
 type RedisCredentials struct {
-	// The username that will be used for authenticating connections to the Redis servers. If not
-	// supplied, "default" will be used.
-	Username string
-	// The password that will be used for authenticating connections to the Redis servers.
-	Password string
+	// Optional: the username that will be used for authenticating connections to the Redis servers. If not supplied, "default"
+	// will be used.
+	Username *string
+	// Required: the password that will be used for authenticating connections to the Redis servers.
+	Password *string
+}
+
+func (credentials *RedisCredentials) validate() error {
+	if credentials.Password == nil {
```

Comment:
shouldn't we check username as well?
I think a flag to do auth is `baseClientConfiguration.credentials != nil`

Reply:
In the Java code, the username is optional. If the user does not supply a username, the Java code does not set a username in the connection request protobuf message. I'm not sure what you mean by "I think a flag to do auth is baseClientConfiguration.credentials != nil", can you clarify?

Reply:
Right, sorry, it is my mistake